rdt/hyper_transformer.py

```diff
@@ -151,20 +156,21 @@ def _field_in_data(field, data):
     def _set_field_sdtype(self, data, field):
         clean_data = data[field].dropna()
         kind = clean_data.infer_objects().dtype.kind
-        self._fitted_field_sdtypes[field] = self._DTYPES_TO_SDTYPES[kind]
+        assert field not in self.field_sdtypes
+        self.field_sdtypes[field] = self._DTYPES_TO_SDTYPES[kind]
 
     def _populate_field_sdtypes(self, data):
         # get set of provided fields including multi-column fields
         provided_fields = set()
-        for field in self.provided_field_sdtypes.keys():
-            self._add_field_to_set(field, provided_fields)
+        for field in self.field_sdtypes.keys():
```

Comment:
We can iterate directly over the `dict` no  needs for `keys()`.

Reply:
Eh, I prefer to be explicit, so I usually use `keys()` when iterating over dicts (except if the line is over 100 characters...)